google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/apis/GoogleCloudApiDetailsPanel.java

```diff
@@ -229,6 +238,52 @@ private void updateUI() {
     managementWarningTextPane.setText(GctBundle.message("cloud.apis.management.section.info.text"));
   }
 
+  /**
+   * Asynchronously fetches and displays the version of the client library that is managed by the
+   * given BOM version.
+   *
+   * @param bomVersion the version of the BOM from which to load the version of the current client
+   *     library
+   */
+  // TODO (eshaul) this unoptimized implementation fetches all managed BOM versions each time the
+  // BOM is updated and library is selected. The bomVersion -> managedLibraryVersions results can be
+  // cached on disk to reduce network calls.
+  @SuppressWarnings("FutureReturnValueIgnored")
+  void updateManagedLibraryVersionFromBom(String bomVersion) {
+    if (currentCloudLibrary.getClients() != null) {
+      CloudLibraryUtils.getFirstJavaClient(currentCloudLibrary)
+          .ifPresent(
+              client -> {
+                CloudLibraryClientMavenCoordinates coordinates = client.getMavenCoordinates();
+
+                if (coordinates != null) {
+                  versionLabel.setIcon(GoogleCloudCoreIcons.LOADING);
+                  versionLabel.setText("");
```

Comment:
What would happen if the following code returns empty optional without the version? Will the version label remain empty and showing `loading`?

Reply:
yeah good catch. if it fails to fetch the version I think we should probably just hide the loader and show a blank version. The reason I think this is better than falling back to the version from the json file is because, in this situation, there _is_ a BOM. So its not 100% essential for us to know the version (its really just for display for the user) since the pom dependency itself won't have a version.

I can update this and add a test to cover this situation. wdyt?

Reply:
updated

Reply:
I agree, good reasoning.